File: examples/example_kam_diagnostics.py
"""Smoke example for sst_kam_diagnostics (Canon v0.8.25+)."""
from __future__ import annotations

import math
import os
import sys

sys.path.insert(0, os.path.abspath("."))
sys.path.insert(0, os.path.join(os.path.abspath("."), "src"))

try:
    from SSTcore import KAMDiagnosticsAPI, KAMSector
except ImportError:
    from sstcore import KAMDiagnosticsAPI, KAMSector

r = KAMDiagnosticsAPI.stage1(
    KAMSector.S,
    frequencies=[1.0, math.sqrt(2.0)],
    hessian_row_major=[2.0, 0.0, 0.0, 3.0],
    diophantine_tau=1e-4,
)
print(
    "stage1:",
    r.status,
    r.achieved_stage,
    "det=",
    r.hessian_determinant,
    KAMDiagnosticsAPI.stage_name(r.achieved_stage),
)

phi = 0.5 * (1.0 + math.sqrt(5.0))
print("golden_ratio_null_test:", KAMDiagnosticsAPI.golden_ratio_null_test(phi), KAMDiagnosticsAPI.golden_ratio_null_test(1.5))
print("sector_name:", KAMDiagnosticsAPI.sector_name(KAMSector.S))

File: examples/example_kam_diagnostics.ts
/**
 * sst_kam_diagnostics_node.cpp — see examples/example_kam_diagnostics.py
 */
const sst = require('../index.js');

function main() {
    if (!sst.isAvailable || typeof sst.kamStage1 !== 'function') {
        console.log('[SKIP] kam_diagnostics: addon not available');
        return;
    }
    const r = sst.kamStage1(true, [1.0, Math.SQRT2], [2, 0, 0, 3], 1e-4);
    console.log('kamStage1:', r.status, r.achievedStage, 'det=', r.hessianDeterminant);

    const phi = 0.5 * (1 + Math.sqrt(5));
    console.log('goldenRatioNullTest:', sst.goldenRatioNullTest(phi), sst.goldenRatioNullTest(1.5));
}

main();
